connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -179,7 +179,7 @@ public static String COMMIT_TASKS_KEY(String connectorName) {
     // converter/serializer changes causing keys to change. We need to absolutely ensure that the keys remain precisely
     // the same.
     public static final Schema CONNECTOR_CONFIGURATION_V0 = SchemaBuilder.struct()
-            .field("properties", SchemaBuilder.map(Schema.STRING_SCHEMA, Schema.OPTIONAL_STRING_SCHEMA))
+            .field("properties", SchemaBuilder.map(Schema.STRING_SCHEMA, Schema.OPTIONAL_STRING_SCHEMA).build())
```

Comment:
@hachikuji There are some weird issues re: self-referencing schemas, which actually do appear in practice sometimes. In that case, having SchemaBuilder implement Schema is actually useful. Building immediately where you are in that call is risky since it doesn't guarantee the user has finished building the field's schema.

We could argue about whether supporting that data is a good idea or not, or we could use a different approach to building them (save the SchemaBuilder and the build out the complete schema later), but I think this is not as trivial a change (only requiring a few overloads) as you're suggesting here. (Also, that change would require a KIP now...)